code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -138,12 +138,40 @@ const styles = {
   }
 };
 
-
 function dotClicked(url, e) {
   e.preventDefault();
   saveAnswersAndNavigate(url);
 }
 
+const BubbleInterior = React.createClass({
```

Comment:
Ah, makes sense.  I'm used to one-class-per-file to make it easy to find things, but I can see the exception when it's only used one place.